provider/cmd/pulumi-gen-eks/main.go

```diff
@@ -374,7 +389,10 @@ func generateSchema() schema.PackageSpec {
 						Description: "The subnets to use for worker nodes. Defaults to the value of subnetIds.",
 					},
 					"clusterSecurityGroup": {
-						TypeSpec: schema.TypeSpec{Ref: awsRef("#/resources/aws:ec2%2FsecurityGroup:SecurityGroup")},
+						TypeSpec: schema.TypeSpec{
+							Ref:   awsRef("#/resources/aws:ec2%2FsecurityGroup:SecurityGroup"),
+							Plain: true,
+						},
```

Comment:
This is plain in the implementation:

https://github.com/pulumi/pulumi-eks/blob/c6f1fce98fa2683abe7ffbad7f166da99928fb83/nodejs/eks/cluster.ts#L1078-L1082